pybaseball/playerid_lookup.py

```diff
@@ -1,26 +1,50 @@
 import pandas as pd
 import requests
 import io
+import os
 
-# dropped key_uuid. looks like a has we wouldn't need for anything. 
-# TODO: allow for typos. String similarity? 
+# dropped key_uuid. looks like a has we wouldn't need for anything.
+# TODO: allow for typos. String similarity?
 
+url = "https://raw.githubusercontent.com/chadwickbureau/register/master/data/people.csv"
+register_file = 'chadwick-register.csv'
+
+def chadwick_register(save: bool = False) -> pd.DataFrame:
+    ''' Get the Chadwick register Database '''
+
+    if os.path.exists(register_file):
+        table = pd.read_csv(register_file)
+        return table
 
-def get_lookup_table():
     print('Gathering player lookup table. This may take a moment.')
-    url = "https://raw.githubusercontent.com/chadwickbureau/register/master/data/people.csv"
-    s=requests.get(url).content
-    table = pd.read_csv(io.StringIO(s.decode('utf-8')), dtype={'key_sr_nfl': object, 'key_sr_nba': object, 'key_sr_nhl': object})
-    #subset columns
-    cols_to_keep = ['name_last','name_first','key_mlbam', 'key_retro', 'key_bbref', 'key_fangraphs', 'mlb_played_first','mlb_played_last']
+    s = requests.get(url).content
+    cols_to_keep = [
+        'name_last', 'name_first', 'key_mlbam', 'key_retro', 'key_bbref', 'key_fangraphs', 'mlb_played_first',
+        'mlb_played_last'
+    ]
+    table = pd.read_csv(io.StringIO(s.decode('utf-8')), usecols=cols_to_keep)
+
+    table.dropna(how='all', subset=cols_to_keep[3:], inplace=True) # Keep only the major league rows
```

Comment:
`cols_to_keep[3:]` could be a potential source of bugs if someone modifies `cols_to_keep`. Would it make sense to store the major league rows and the other rows in separate arrays? You can join them in places where you use `cols_to_keep`, and pass `major_league_rows` as the argument for `subset`

Reply:
Addressed by creating a `mlb_only_cols` there and using that for the subset and for building `cols_to_keep`